app/models/state_file_w2.rb

```diff
@@ -36,14 +36,19 @@ class StateFileW2 < ApplicationRecord
   XML
   belongs_to :state_file_intake, polymorphic: true
 
-  validates :w2_index, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates_numericality_of :w2_index, only_integer: true, message: I18n.t('errors.messages.whole_number')
+  validates :w2_index, presence: true, numericality: { greater_than_or_equal_to: 0 }
 
   validates :employer_state_id_num, format: { with: /\A(\d{0,17})\z/ }
-  validates :state_wages_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, if: -> { state_wages_amt.present? }
-  validates :state_income_tax_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, if: -> { state_income_tax_amt.present? }
-  validates :local_wages_and_tips_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, if: -> { local_wages_and_tips_amt.present? }
-  validates :local_income_tax_amt, numericality: { only_integer: true, greater_than_or_equal_to: 0 }, if: -> { local_income_tax_amt.present? }
-  validates :locality_nm, presence: { message: -> (_object, _data) { I18n.t('state_file.questions.ny_w2.edit.locality_nm_missing_error') } }, if: -> { local_wages_and_tips_amt.present? && local_wages_and_tips_amt.positive? }
+  validates_numericality_of :state_wages_amt, only_integer: true, message: I18n.t('errors.messages.whole_number'), if: -> { state_wages_amt.present? }
+  validates :state_wages_amt, numericality: { greater_than_or_equal_to: 0 }, if: -> { state_wages_amt.present? }
+  validates_numericality_of :state_income_tax_amt, only_integer: true, message: I18n.t('errors.messages.whole_number'), if: -> { state_income_tax_amt.present? }
```

Comment:
I see you have a pattern of two validations per field.  Is it possible to have one that achieves the desired result? It seems we are looking for these things:

- that the number be positive
- that the number be an integer 
- to display the error message otherwise

Have you tried something like:

`validates :your_number_attribute, numericality: { greater_than_or_equal_to: 0, only_integer: true, message: I18n.t('errors.messages.whole_number') } , if: -> { your_number_attribute.present? }`

Reply:
i tried doing this but it would show up as an error for both greater_than_or_equal_to: 0 and only_integer: true

Reply:
So the validation only fails if both errors are present this way?

Reply:
yeah the message becomes the message for both validations

Reply:
Like using the code you gave would not give the right message if someone put 0 for greater than or equal to 0
<img width="488" alt="Screenshot 2024-03-11 at 4 07 26 PM" src="https://github.com/codeforamerica/vita-min/assets/79296603/ea0b0bcb-5d13-47c4-96e7-06a8121785f2">

Reply:
Makes sense! Tnx for clarifying